neurons/validator.py

```diff
@@ -696,16 +699,22 @@ def receive_checkpoint(self, synapse: template.protocol.ValidatorCheckpoint) ->
 
             error_message = ""
             try:
-                # get our current checkpoint
-                now_ms = TimeUtil.now_in_millis()
-                checkpoint_dict = generate_request_core(time_now=now_ms)
-
-                # compress json and encode as base64 to keep as a string
-                checkpoint_str = json.dumps(checkpoint_dict, cls=CustomEncoder)
-                compressed = gzip.compress(checkpoint_str.encode("utf-8"))
-                encoded_checkpoint = base64.b64encode(compressed).decode("utf-8")
-
-                synapse.checkpoint = encoded_checkpoint
+                # reset checkpoint every 10 mins
+                if TimeUtil.now_in_millis() - self.last_checkpoint_time > 1000 * 60 * 10:
```

Comment:
This logic should be contained within the lock to prevent a race condition where one thread wipes out self.encoded_checkpoint right after it is updated.

with self.checkpoint_lock:
if TimeUtil.now_in_millis() - self.last_checkpoint_time > 1000 * 60 * 10:
self.encoded_checkpoint = ""
 if not self.encoded_checkpoint:
  self.last_checkpoint_time = TimeUtil.now_in_millis()
  ...